src/quest_manager/views.py

```diff
@@ -975,6 +975,9 @@ def complete(request, submission_id):
         if "complete" not in request.POST and "comment" not in request.POST:
             raise Http404("unrecognized submit button")
 
+        if not submission.draft_comment:  # if no comment, the quest is not complete
```

Comment:
Also, how would this happen?

Reply:
My comment was a poor choice of words. It wasn't to check if the submission was complete, but rather to ensure the user had actually gone into the `submission` view. And as mentioned for the check I made in the `ajax_save_draft` function, this would only happen if the system was improperly configured, or if a student used a headless web browser.

I could create a unit test that did this by creating a submission with no draft_comment value, then trying to pass it into the `complete` view. But if you think this check is redundant, let me know and I'll remove it.

Reply:
```py
def test_ajax_save_draft_no_submission_draft_comment(self):
        """If a draft comment is not associated with the submission, then there is no submission in progress,
        so return 404"""
        self.client.force_login(self.test_student1)
        quest = baker.make(Quest, name="TestSaveDrafts")
        # create a submission with no draft comment
        sub = baker.make(QuestSubmission, draft_comment=None,
                         quest=quest)

        response = self.client.post(
            reverse('quests:ajax_save_draft'),
            data={
                'comment': "I am a test",
                'submission_id': sub.id,
            },
            HTTP_X_REQUESTED_WITH='XMLHttpRequest',
        )
        self.assertEqual(response.status_code, 404)
```